code/game/objects/effects/glowshroom.dm

```diff
@@ -35,7 +38,10 @@ var/list/blacklisted_glowshroom_turfs = typecacheof(list(
 
 /obj/structure/glowshroom/New()
 	..()
-	set_light(round(potency/10))
+	addtimer(CALLBACK(src, .proc/setup), 0) //run this on the next tick so we can set vars first
+
+/obj/structure/glowshroom/proc/setup()
```

Comment:
Doesn't work, smart guy; it can't set the vars before set_light and the addtimer is called.